cmake/alpakaCommon.cmake

```diff
@@ -234,20 +235,22 @@ endif()
 
 #-------------------------------------------------------------------------------
 # Find OpenMP.
-if(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE OR ALPAKA_ACC_CPU_BT_OMP4_ENABLE)
+if(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE OR ALPAKA_ACC_ANY_BT_OMP5_ENABLE)
     find_package(OpenMP)
 
     if(OpenMP_CXX_FOUND)
-        if(ALPAKA_ACC_CPU_BT_OMP4_ENABLE)
+        if(ALPAKA_ACC_ANY_BT_OMP5_ENABLED)
             if(OpenMP_CXX_VERSION VERSION_LESS 4.0)
-                message(FATAL_ERROR "OpenMP 4.0 is required but not supported!")
+                message(FATAL_ERROR "ALPAKA_ACC_ANY_BT_OMP5_ENABLE requires compiler support for OpenMP at lease 4.0, 5.0 is recommended.")
+            elseif(OpenMP_CXX_VERSION VERSION_LESS 5.0)
+                message(WARNING "OpenMP < 5.0, for ALPAKA_ACC_ANY_BT_OMP5_ENABLE 5.0 is recommended.")
             endif()
         endif()
 
         target_link_libraries(alpaka INTERFACE OpenMP::OpenMP_CXX)
 
-        # Clang versions starting from 3.9 support OpenMP 4.0 only when given the corresponding flag
-        if(ALPAKA_ACC_CPU_BT_OMP4_ENABLE)
+        # Clang versions starting from 3.9 support OpenMP 5.0 only when given the corresponding flag
+        if(ALPAKA_ACC_ANY_BT_OMP5_ENABLE)
             target_link_options(alpaka INTERFACE $<$<CXX_COMPILER_ID:AppleClang,Clang>:-fopenmp-version=40>)
```

Comment:
Do we need to use `-fopenmp-version=50`? But which clang versions support this?

Reply:
I know that clang 11 claims to support OpenMP 5.0 (it is incomplete), I did not check where exactly clang 10 is at.

This PR is not explicitly using OpenMP 5.0, but it relies on the compiler being lenient with some of OpenMP < 5.0's restrictions, chiefly the `static constexpr` member thing: https://github.com/alpaka-group/alpaka/pull/1118 .

Reply:
The Math test uses `static constexpr` members, which is why it is disabled for GCC.